github/GithubObject.py

```diff
@@ -92,6 +93,26 @@ def remove_unset_items(data: Dict[str, Any]) -> Dict[str, Any]:
 Opt = Union[T, _NotSetType]
 
 
+def is_defined(v: Union[T, _NotSetType]) -> TypeGuard[T]:
```

Comment:
What about
```suggestion
def is_set(v: Union[T, _NotSetType]) -> TypeGuard[T]:
```

Reply:
`set` is a python builtin type, maybe confusing.

Reply:
Well, this holds for `NotSet` and `_NotSetType` as well.

Let's stick to `defined` then.

Reply:
maybe rename `NotSet` to `Undefined`

Reply:
Like that, but in a separate PR, will touch all files :-/